From a3d9201bff675e5ad2a85391a754fd9f3d88065b Mon Sep 17 00:00:00 2001
From: Zubair Lutfullah Kakakhel <zubair@resin.io>
Date: Wed, 19 Sep 2018 11:34:34 +0100
Subject: [PATCH] rpi.h: Add bootlimit and altbootcmd to environment

When bootlimit=1, if bootcount exceeds 1, env_resin.h will return
the alternate root partition when resin_set_kernel_root is run.

When bootlimit=1, u-boot expects altbootcmd to be present. We handle
logic in env_resin.h, so just run normal bootcmd in altbootcmd as well

Signed-off-by: Zubair Lutfullah Kakakhel <zubair@resin.io>
Upstream-Status: Inappropriate [configuration]
---
 include/configs/rpi.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index a97550b..f6fc739 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -157,6 +157,8 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"dhcpuboot=usb start; dhcp u-boot.uimg; bootm\0" \
+	"bootlimit=1\0" \
+	"altbootcmd=run bootcmd\0" \
 	ENV_DEVICE_SETTINGS \
 	ENV_MEM_LAYOUT_SETTINGS \
 	BOOTENV
-- 
2.7.4

